src/Sulu/Bundle/ContentBundle/Controller/NodeController.php

```diff
@@ -592,16 +592,14 @@ public function putAction(Request $request, $uuid)
         $language = $this->getLanguage($request);
         $type = $request->query->get('type', 'page');
 
-        try {
-            $document = $this->getDocumentManager()->find(
-                $uuid,
-                $language,
-                ['type' => $type, 'load_ghost_content' => false]
-            );
-        } catch (DocumentNotFoundException $e) {
-            $e = new EntityNotFoundException(PageDocument::class, $uuid, $e);
+        $document = $this->getDocumentManager()->find(
+            $uuid,
+            $language,
+            ['load_ghost_content' => false]
+        );
 
-            return $this->handleView($this->view($e->toArray(), 404));
+        if (!$this->get('sulu_hash.request_hash_checker')->checkHash($request, $document)) {
+            throw new InvalidHashException(get_class($document), $document->getUuid());
```

Comment:
Wanted to put the throwing of this exception into the `checkHash` function, but this is not really possible, because there is no standardized way to get the ID or UUID of the object/document to check. /cc @chirimoya